src/viam/ur/module/ur_arm_state_disconnected.cpp

```diff
@@ -12,23 +12,53 @@ std::string URArm::state_::state_disconnected_::describe() const {
 }
 
 std::chrono::milliseconds URArm::state_::state_disconnected_::get_timeout() const {
-    return std::chrono::seconds(1);
+    // If we have a pending connection, we are polling for completion,
+    // so increase the sampling rate. Otherwise we want to wait a more
+    // reasonable length of time before attempting to connect again.
+    return pending_connection ? std::chrono::milliseconds(5) : std::chrono::seconds(1);
 }
 
 std::optional<URArm::state_::event_variant_> URArm::state_::state_disconnected_::recv_arm_data(state_&) {
     return std::nullopt;
 }
 
 std::optional<URArm::state_::event_variant_> URArm::state_::state_disconnected_::upgrade_downgrade(state_& state) {
-    reconnect_attempts++;
-    const auto log_at_n_attempts = 100;
-    if (reconnect_attempts % log_at_n_attempts == 0) {
-        VIAM_SDK_LOG(info) << "disconnected: attempting recovery";
+    if (pending_connection) {
+        // There is a pending connection attempt. Poll it for
+        // completion. If it has completed, get the result, which will
+        // either cause an exception to be thrown or return the arm
+        // connection that we can wrap up into a connection established
+        // event. If the future isn't ready, there is no event to
+        // return, and we will come back to poll again.
+        switch (pending_connection->wait_for(std::chrono::seconds(0))) {
+            case std::future_status::ready: {
+                return event_connection_established_{std::exchange(pending_connection, std::nullopt)->get()};
+            }
+            case std::future_status::timeout: {
+                break;
+            }
+            case std::future_status::deferred: {
```

Comment:
are these the only cases? do we not need a default?

Reply:
These cases are exhaustive so no default is needed. If std::future_status ever grew a new value, the compiler or linter would almost certainly whine at us. In fact, I'd have preferred to not write one for `deferred` at all, but if I don't, I get a tidy warning:
```
/src/src/viam/ur/module/ur_arm_state_disconnected.cpp:33:17: error: enumeration value 'deferred' not handled in switch [-Werror,-Wswitch]
   33 |         switch (pending_connection->wait_for(std::chrono::seconds(0))) {
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```